demo/js/demo-animated-trigger.js

```diff
@@ -0,0 +1,79 @@
+const { useState } = require('react');
+const React = require('react');
+const ReactDOM = require('react-dom');
+const FocusTrap = require('../../dist/focus-trap-react');
+
+const container = document.getElementById('demo-animated-trigger');
+
+const focusTrapOptions = {
+  // There is a delay between when the class is removed
+  // and when the trigger is focusable
+  checkCanReturnFocus: (triggerButton) => {
+    return new Promise((resolve) => {
+      const interval = setInterval(() => {
+        if (getComputedStyle(triggerButton).visibility !== 'hidden') {
+          resolve();
+          clearInterval(interval);
+        }
+      }, 5);
+    });
+  },
+  // Called after focus is sent to the trigger button
+  onPostDeactivate: () => {
+    // eslint-disable-next-line no-console
+    console.log(
+      'Focus has been sent to the animated focus trap trigger button'
+    );
+  },
+};
+
+const DemoAnimatedTrigger = () => {
+  const [isTrapActive, setIsTrapActive] = useState(false);
+
+  return (
+    <>
+      <style>{`
```

Comment:
Same as above, the styles for the animated demos are an important part of the demo so should be displayed as part of the demo.